packages/app-desktop/gui/NoteEditor/NoteBody/CodeMirror/CodeMirror.tsx

```diff
@@ -731,15 +731,22 @@ function CodeMirror(props: NoteBodyEditorProps, ref: any) {
 	// It might be buggy, refer to the below issue
 	// https://github.com/laurent22/joplin/pull/3974#issuecomment-718936703
 	useEffect(() => {
-		function pointerInsideEditor(x: number, y: number) {
+		function pointerInsideEditor(params: any) {
+			const x = params.x, y = params.y, isEditable = params.isEditable, inputFieldType = params.inputFieldType;
 			const elements = document.getElementsByClassName('codeMirrorEditor');
-			if (!elements.length) return null;
+
+			// To check if codeMirrorEditor doesn't exist. --OR--
+			// If isEditable is false which means there is a blur layout on codeMirrorEditor (e.g. add hyperlink)
+			// or mouse cursor is out of codeMirrorEditor. --OR--
+			// If the context menu was invoked on an input field (isEditable: true) and the input field on codeMirrorEditor,
+			// to avoid context-menu flickering (two context-menu) by checking if is an input field or not by using inputFieldType (e.g plainText, password).
+			if (!elements.length || !isEditable || inputFieldType !== 'none') return null;
```

Comment:
Ok but you didn't need to write that much text to explain this line. Something like `!isEditable` is self-explanatory, and we know that `||` is "OR".

What needs to be explained is `inputFieldType !== 'none'`. So that means the CodeMirror always have an "inputFieldType" of "none"? Could you explain why? And are you sure it's really always "none"?

Reply:
hmmm🤔 ok, I will re-comment according to inputFieldType only.

+ CodeMirror always have an "inputFieldType" of "none"? Yes

+ why? Because the input field type of CodeMirror is "textarea" so the inputFieldType = "none",
and any single-line input above codeMirror has inputFieldType value according to the type of input e.g.(text = plainText, password = password, ...).